### PR TITLE
Newhint#17 & Newhint#18 imports of SECP256R1_ALPHA and SECP256R1_N

### PR DESCRIPTION
**Pull request by mfachal**
# TITLE

## Description

Add hints #17 and #18: IMPORT_SECP256R1_ALPHA and IMPORT_SECP256R1_N

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [ ] CHANGELOG has been updated.

### CODE REVIEW

**Review by Oppen — APPROVED** [review]

File: cairo_programs/secp256r1_div_mod_n.cairo [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    return (res=res);
+}
+
+func main{range_check_ptr}(){
+    let x = BigInt3(235, 522, 111);
+    let y = BigInt3(1323, 15124, 796759);
+
+    let a = div_mod_n(x, y);
+    assert a = a;
```
[the remark this answers is not in the document]
Comment (Oppen) [reply cairo_programs/secp256r1_div_mod_n.cairo:41]: I think this is only to force the VM to evaluate it. I've seen this pattern a few times.

**Review by Oppen — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -879,4 +892,4 @@
         * `pub fn from_vm_error(runner: &CairoRunner, error: VirtualMachineError, pc: usize) -> Self` is now `pub fn from_vm_error(runner: &CairoRunner, vm: &VirtualMachine, error: VirtualMachineError) -> Self`
         * `pub fn get_location(pc: &usize, runner: &CairoRunner) -> Option<Location>` is now `pub fn get_location(pc: usize, runner: &CairoRunner) -> Option<Location>`
         * `pub fn decode_instruction(encoded_instr: i64, mut imm: Option<BigInt>) -> Result<instruction::Instruction, VirtualMachineError>` is now `pub fn decode_instruction(encoded_instr: i64, mut imm: Option<&BigInt>) -> Result<instruction::Instruction, VirtualMachineError>`
-        * `VmExcepion` field's string format now mirror their cairo-lang conterparts.
\ No newline at end of file
+        * `VmExcepion` field's string format now mirror their cairo-lang conterparts.
```
Comment (juanbono) [inline CHANGELOG.md:895]: typo in VmException

Comment (juanbono) [reply CHANGELOG.md:895]: and in Counterparts

File: src/hint_processor/builtin_hint_processor/secp/secp_utils.rs [hunk 3]
```diff
@@ -48,7 +48,7 @@ lazy_static! {
         "115792089210356248762697446949407573530086143415290314195533631308867097853951"
     ).unwrap();
     //SECP256R1_N = 0xFFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551
-    pub(crate) static ref SECP256R1_N: BigUint = BigUint::from_str(
+    pub(crate) static ref SECP256R1_N: BigInt = BigInt::from_str(
```
[the remark this answers is not in the document]
Comment (juanbono) [reply src/hint_processor/builtin_hint_processor/secp/secp_utils.rs:51]: yeah why this?

**Review by juanbono — CHANGES_REQUESTED** [review]
Please review @fmoletta comment

File: src/hint_processor/builtin_hint_processor/secp/secp_utils.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (mfachal) [reply src/hint_processor/builtin_hint_processor/secp/secp_utils.rs:51]: Yeah we discussed it later on; thing is it is used later on in the div_mod function, which takes a BigInt, and changing that would mean changing many other hints; this same thing happened with the secp256r1_p hint (#1014) - I wasn't sure if I should go around touching every usage of div_mod everywhere.

File: CHANGELOG.md (hunk 2 above)
Comment (Oppen) [reply CHANGELOG.md:895]: This is fixed in another PR. Just merge main and it should be OK.

File: src/hint_processor/builtin_hint_processor/secp/secp_utils.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (juanbono) [reply src/hint_processor/builtin_hint_processor/secp/secp_utils.rs:51]: Makes sense. Thanks :)

**Review by juanbono — APPROVED** [review]

**Review by fmoletta — APPROVED** [review]

**Pull request merged**